src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,229 @@
+package cacophony.grammar.syntaxtree
+
+import cacophony.utils.Location
+
+sealed class Expression(val range: Pair<Location, Location>) { // everything in cacophony is an expression
+    class Variable(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    // should Type also inherit expression? if so, can we do for example "; Bool;" in our language?
+    class Type(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableDeclaration(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableTypedDeclaration(
+        range: Pair<Location, Location>,
+        variable: VariableDeclaration,
+        type: Type,
+    ) : Expression(range)
+
+    class Literal(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class Keyword(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        class KeywordReturn(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordBreak(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+    }
+
+    // expression in parenthesis
+    class NestedExpression(
+        range: Pair<Location, Location>,
+        val expression: Expression,
+    ) : Expression(range)
+
+    // series of expressions separated by semicolons, returning value of the last one
+    // we agreed that (x; y;) returns Unit - should we create artificial class for empty expression after the last semicolon,
```

Comment:
imo we should, it would just be an expression with empty range, based on that we could immediately deduce that its value is `Unit`